core/src/main/java/org/arquillian/smart/testing/Logger.java

```diff
@@ -145,6 +172,10 @@ public void finest(String msg, Object... args) {
         jul.finest(getFormattedMsg(msg, args));
     }
 
+    public Boolean isDebugEnabled() {
+        return Boolean.valueOf(System.getProperty(SMART_TESTING_DEBUG, "false"));
```

Comment:
As @MatousJobanek pointed out in another PR, if a property is not set it will return `null` `Boolean.valueOf` will make it `false`. But I believe this verbosity here clearly explains what we expect as a default

Reply:
If I remember correctly we also wanted to have it enabled when user run the build with `mvn -X`

Reply:
Yes but I think it is better as you said to have in this way here to be clear what is the default.

Reply:
Yeah, default is fine. But what about `-X` flag?

Reply:
Do you think is possible to get it from there?

Reply:
@bartoszmajsak, it doesn't work for the `-X` flag at the moment. `-X` is not set as system or user property. It is a maven option and I am not sure how I can get this value from the build.

In the test-bed, we can set it with the `builderOptions`, but I don't know how to retrieve it from the 
maven configurer.

Reply:
After a quick look into Maven Core source code I can think about one way of doing it... Which in fact  reveals that we can rethink our logging approach. In the `SmartTestingMavenConfigurer` you can inject Maven logger as a field:

```
@Requirement
private org.codehaus.plexus.logging.Logger log;
```

and this gives you access to `log.isDebugEnabled()` which is true when `-X` is set.

Reply:
The thing is that then we log `core` elements with one logger and maven things with another,  not bad but just noticing it

Reply:
The question is however - how to propagate this information further down the execution.

Reply:
> The thing is that then we log core elements with one logger and maven things with another, not bad but just noticing it

Sure, rethinking part is open for discussion. The access to this information (if we are in debug mode in Maven) is another thing.

Reply:
Maybe we can just use this logger instance to detect if `-X` is set

Reply:
On the other hand, this got me thinking. As `-X` is a command line option which is a JVM thing.

```java
RuntimeMXBean runtimeMxBean = ManagementFactory.getRuntimeMXBean();
List<String> arguments = runtimeMxBean.getInputArguments()
```
and this list contains `-Xdebug` element. That's all what we need.

Reply:
Ok, it seems that `-X` != `-Xdebug`. Makes me wonder how Mave reads this flag. @MatousJobanek any idea?

Reply:
The maven `org.codehaus.plexus.logging.Logger` is available for the plugins, but it is not available nor used in any `SurefireProvider` implementation. In the surefire providers, they are using their own implementation (based on the information if the execution is in a fork or not). But in any case, they log on stdout. This is also why I have implemented it using stdout and stderr.
There is a way to get the instance of the Surefire logger from the `ProviderParameters`, but this will work only for maven environment, not for IDEs.

Reply:
We need an instance of `CliRequest`